neural_tangents/stax.py

```diff
@@ -1447,6 +1542,183 @@ def Conv(out_chan, filter_shape, strides=None, padding=Padding.VALID.name,
+
+    return norm * lax.conv_transpose(
+        inputs,
+        W,
+        strides,
+        apply_padding.name,
+        dimension_numbers=dimension_numbers) + b_rescale * b
+
+  def kernel_fn(kernels):
+    """Compute the transformed kernels after a conv layer."""
+    var1, nngp, var2, ntk, is_height_width, marginal, cross = (
+        kernels.var1, kernels.nngp, kernels.var2, kernels.ntk,
+        kernels.is_height_width, kernels.marginal, kernels.cross)
+
+    if cross > M.OVER_PIXELS and not is_height_width:
+      filter_shape_nngp = filter_shape[::-1]
+      strides_nngp = strides[::-1]
+    else:
+      filter_shape_nngp = filter_shape
+      strides_nngp = strides
+
+    if cross == M.OVER_PIXELS:
+
+      def conv_nngp_unscaled(x):
+        if _is_array(x):
+          x = _conv_nngp_4d(x, filter_shape_nngp, strides_nngp, padding, transpose=True)
+        return x
+    elif cross in [M.OVER_POINTS, M.NO]:
+
+      def conv_nngp_unscaled(x):
+        if _is_array(x):
+          x = _conv_nngp_5or6d_double_conv(x, filter_shape_nngp,
+                                           strides_nngp, padding, transpose=True)
+        return x
+
+      is_height_width = not is_height_width
+    else:
+      raise NotImplementedError(
+          "Only implemented for `OVER_PIXELS`, `OVER_POINTS` and `NO`;"
+          " supplied {}".format(cross))
+
+    def conv_nngp(x):
+      x = conv_nngp_unscaled(x)
+      return _affine(x, W_std, b_std)
+
+    if marginal == M.OVER_PIXELS:
+      def conv_var(x):
+        x = _conv_var_3d(x, filter_shape_nngp, strides_nngp, padding, transpose=True)
+        x = _affine(x, W_std, b_std)
+        return x
+    elif marginal in [M.OVER_POINTS, M.NO]:
+      def conv_var(x):
+        if _is_array(x):
+          x = _conv_nngp_5or6d_double_conv(x, filter_shape_nngp,
+                                           strides_nngp, padding, transpose=True)
+        x = _affine(x, W_std, b_std)
+        return x
+    else:
+      raise NotImplementedError(
+          "Only implemented for `OVER_PIXELS`, `OVER_POINTS` and `NO`;"
+          " supplied {}".format(marginal))
+
+    var1 = conv_var(var1)
+    var2 = conv_var(var2)
+
+    if parameterization == 'ntk':
+      nngp = conv_nngp(nngp)
+      ntk = conv_nngp(ntk) + nngp - b_std**2 if ntk is not None else ntk
+    elif parameterization == 'standard':
+      nngp_unscaled = conv_nngp_unscaled(nngp)
+      if ntk is not None:
+        ntk = (
+            input_total_dim(kernels.shape1) * nngp_unscaled + 1. +
+            W_std**2 * conv_nngp_unscaled(ntk))
+      nngp = _affine(nngp_unscaled, W_std, b_std)
+
+    return kernels._replace(
+        var1=var1, nngp=nngp, var2=var2, ntk=ntk, is_gaussian=True,
+        is_height_width=is_height_width, marginal=marginal, cross=cross)
+
+  setattr(kernel_fn, _COVARIANCES_REQ, {'marginal': M.OVER_PIXELS,
+                                        'cross': M.OVER_PIXELS})
+  return init_fn, apply_fn, kernel_fn
```

Comment:
This method appears to have quite a lot of similarities with `_GeneralConv` - do you think it would be possible to merge them together into one method with a `transpose` parameter? (if not, perhaps still extract some common functionality into a third method)

Reply:
Thanks a lot for adding ConvTranspose.  
I agree with Roman to merge Conv and ConvTranspose into one function if possible. Could you also add unit test for ConvTranpose? 

Thanks

Reply:
@romanngg I combined then into one function.